### PR TITLE
[SPARK-26151][SQL][FOLLOWUP] Return partial results for bad CSV records

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

Updated SQL migration guide according to changes in https://github.com/apache/spark/pull/23120

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan Please, have a look at the PR.

File: docs/sql-migration-guide-upgrade.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ displayTitle: Spark SQL Upgrading Guide
 
   - Since Spark 3.0, CSV datasource uses java.time API for parsing and generating CSV content. New formatting implementation supports date/timestamp patterns conformed to ISO 8601. To switch back to the implementation used in Spark 2.4 and earlier, set `spark.sql.legacy.timeParser.enabled` to `true`.
 
+  - In Spark version 2.4 and earlier, CSV datasource converts a malformed CSV string to a row with all `null`s in the PERMISSIVE mode if specified schema is `StructType`. Since Spark 3.0, returned row can contain non-`null` fields if some of CSV column values were parsed and converted to desired types successfully.
```
Comment (HyukjinKwon) [inline docs/sql-migration-guide-upgrade.md]: Ah, `from_csv` and `to_csv` are added in 3.0 so it's intentionally not mentioned. BTW, I think CSV functionalities can only have `StructType` so maybe we don't have to mention.

**Review by HyukjinKwon — APPROVED** [review]

File: docs/sql-migration-guide-upgrade.md (hunk 1 above)
Comment (MaxGekk) [reply docs/sql-migration-guide-upgrade.md]: you are right. I will remove the part about `StructType`

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**